File: Windows/FireflyDevice/FDFireflyIceSimpleTask.cpp
//
//  FDFireflyIceSimpleTask.cpp
//  FireflyDevice
//
//  Created by Denis Bohm on 10/17/13.
//  Copyright (c) 2013-2014 Firefly Design LLC / Denis Bohm. All rights reserved.
//

#include "FDFireflyIceSimpleTask.h"

namespace FireflyDesign {

	FDFireflyIceSimpleTask::FDFireflyIceSimpleTask(std::shared_ptr<FDFireflyIce> fireflyIce, std::shared_ptr<FDFireflyIceChannel> channel, std::function<void(void)> block)
		: FDFireflyIceTaskSteps(fireflyIce, channel)
	{
		_block = block;
	}

	void FDFireflyIceSimpleTask::executorTaskStarted(FDExecutor *executor) {
		FDFireflyIceTaskSteps::executorTaskStarted(executor);
		_block();
		next(std::bind(&FDFireflyIceSimpleTask::complete, this));
	}

	void FDFireflyIceSimpleTask::complete() {
		done();
	}

}

File: Android/Libraries/fireflydevice/src/main/java/com/fireflydesign/fireflydevice/FDFireflyIceSimpleTask.java
//
//  FDFireflyIceSimpleTask.cpp
//  FireflyDevice
//
//  Created by Denis Bohm on 10/17/13.
//  Copyright (c) 2013-2014 Firefly Design LLC / Denis Bohm. All rights reserved.
//

package com.fireflydesign.fireflydevice;

public class FDFireflyIceSimpleTask extends FDFireflyIceTaskSteps {

    public interface Delegate {
        void run();
    }

    Delegate delegate;

	public FDFireflyIceSimpleTask(FDFireflyIce fireflyIce, FDFireflyIceChannel channel, Delegate delegate) {
		super(fireflyIce, channel);
		this.delegate = delegate;
	}

	public void executorTaskStarted(FDExecutor executor) {
		super.executorTaskStarted(executor);
		delegate.run();
		next("complete");
	}

	void complete() {
		done();
	}

}
